codemp/rd-rend2/tr_glsl.cpp

```diff
@@ -339,11 +339,20 @@ static size_t GLSL_GetShaderHeader(
 						0x0104/* GL_ADD */,
 						GL_REPLACE));
 
+				// [Monsterovich] The shader string is fucked up. I got this from error.
```

Comment:
Don't need any of the changes in this file. The string building itself is fine, it's just printed badly. Looks like this is the culprit here: https://github.com/Monsterovich/OpenJK/blob/027f9a27fc3d29992d17644c4423bb227773ac4e/codemp/rd-rend2/tr_glsl.cpp#L173

I had to split up the shader string because it's too long to fit in the output buffer. If you remove the new line it should print properly. Make sure to print a new line after the loop too :)

Reply:
>The string building itself is fine, it's just printed badly.

No. It's printed okay because it didn't compile the shader before I changed these lines.

Reply:
The printed output you see in the console is *not* the same as the shader string that's passed to OpenGL. The string passed to OpenGL doesn't have the odd line breaks in the middle of words, this is purely a output issue.

If it's fixing the compile issues then it's because the symptoms have been removed, but the root case of the problem is still somewhere.